src/Microsoft.ML.Data/DataLoadSave/MultiFileSource.cs

```diff
@@ -45,14 +45,10 @@ public MultiFileSource(params string[] paths)
             }
 
             List<string> concatenated = new List<string>();
-            if (paths != null)
-            {
-                foreach (string path in paths)
-                    foreach (string rPath in StreamUtils.ExpandWildCards(path))
-                        concatenated.Add(rPath);
-            }
-            else
-                concatenated = null;
+
+            foreach (string path in paths)
+                foreach (string rPath in StreamUtils.ExpandWildCards(path))
+                    concatenated.Add(rPath);
```

Comment:
We shouldn't make this change. `paths` can still be set to null if the user provides that as the only value.

Reply:
OK, I have reverted this file